app/server.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Main where
+
+import qualified Data.Text.Encoding as Text
+import qualified Data.Aeson as Aeson
+import qualified Data.Foldable as Foldable
+import qualified Data.ByteString as ByteString
+import qualified Data.CaseInsensitive as CI
+import Data.List
+import System.Environment
+import Snap.Http.Server
+import Snap.Internal.Http.Server.Config as SnapCfg
+import Snap.Core
+
+type Header = (CI.CI ByteString.ByteString, ByteString.ByteString)
+
+main :: IO ()
+main = do
+  args <- getArgs
+  let port = case args of
+        [pStr] -> read pStr :: Int
+        _      -> 8080
+  httpServe (cfg port) go
+  where
+    cfg port = SnapCfg.defaultConfig
+          { port = Just port
+          , bind = Just "127.0.0.1"
+          }
+
+go :: Snap ()
+go = route [("greet/:name",greeter)
+           ,("headers",echoHeaders)
```

Comment:
Why have a separate route that prints headers? Would it be better to instead have all endpoints print their headers (behind a flag/env-var), using `wrapSite`?

Reply:
I'm working on adding support for headers (as a stepping stone to authentication) and I wanted to have an endpoint that would only print the headers, so that we can keep that testing orthogonal to the behavior of any other routes.